src/PowerShellEditorServices/Workspace/Workspace.cs

```diff
@@ -685,7 +685,15 @@ public static string ConvertPathToDocumentUri(string path)
             }
 
             // ' is not always encoded.  I've seen this in Windows PowerShell.
-            return docUriStrBld.Replace("'", "%27").Insert(0, fileUriPrefix).ToString();
+            docUriStrBld.Replace("'", "%27");
+
+            // Insert /// unless path is a UNC path in which case the proper URI form is file://server/share.
+            if ((docUriStrBld.Length < 2) || ((docUriStrBld[0] != '/') && (docUriStrBld[1] != '/')))
+            {
+                docUriStrBld.Insert(0, "///");
+            }
+
+            return docUriStrBld.Insert(0, fileUriPrefix).ToString();
```

Comment:
Not crucial, but `Insert` is an O(n) operation for a StringBuilder I think, the same as `str1 + str2`, since it has to move the whole string up the length of the prefix in the backing array. Not guaranteed to allocate perhaps. Just want to note it, since we've gone to the effort of using a stringbuilder here.